packages/flame/lib/src/ai/steer/steerable.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2021 Blue Fire team, MIT license
+//
+// -- Original license notice: -------------------------------------------------
+// Copyright 2014 See AUTHORS file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+// TRANSLATED INTO DART WITH MODIFICATIONS from
+// $GDX/ai/steer/Steerable.java
+// -----------------------------------------------------------------------------
+
+import 'package:vector_math/vector_math_64.dart';
+
+import '../utils/location.dart';
+import 'limiter.dart';
+
+/// A [Steerable] is a [Location] that gives access to the character's data
+/// required by steering system.
+///
+/// Notice that there is nothing to connect the direction that a Steerable is
+/// moving and the direction it is facing. For instance, a character can be
+/// oriented along the x-axis but be traveling directly along the y-axis.
+abstract class Steerable extends Location with Limiter {
```

Comment:
This is a strange class hierarchy, `Steerable` being a type of `Location` sounds strange to me at least. I guess this is from the ported library and not something you came up with? It could have `implements Location` instead maybe?

Reply:
Originally all of them are interfaces, so `Steerable` simply extends both `Location` and `Limiter`

Reply:
Then why isn't the implements keyword used, instead of extends?

Reply:
We could make `Steerable` implement `Location` and `Limiter` -- but that would require copy-pasting code from those classes into here. Another alternative is to use both Location and Limiter as mixins, so that it becomes `abstract class Steerable with Location, Limiter`. WDYT, which way is better?

Reply:
Yeah, I think the mixin way is definitely better.